### DISCUSSION TITLE
Scheduled jobs

### DISCUSSION THREAD

**Discussion by andreas-bergstrom** [pocketbase/pocketbase#2592] [Q&A]
I had a look at the current issues and the roadmap but couldn't find anything similar.

Even in small hobby projects there's usually a need to run some business logic on a pre-defined schedule, detached from any request. For Pocketbase this wouldn't have to be raw Go-code, but DB-operations and http-calls would probably be enough.

While this could be achieved through using it as a framework, having it built-in would be superb 👌.

I'd be open to look into this if we get the discussion going.

**Comment by ganigeorgiev** [comment]
This was requested before but I don't think we can have this builtin because it'll always require writing some code that describes what the scheduled job do (making db calls to aggregate some data, sending emails, fetching data from external APIs, manual file operations, etc.). This eventually in the future may become possible if we endedup implementing a "workflow builder" in the Admin UI, but for now [other tasks](https://github.com/orgs/pocketbase/projects/2) are with higher priority.

At the moment the recommend approach is to [use PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) (_better documentation and JS app hooks support is planned for the next v0.17/v0.18 release_) or use the built-in OS cron utils.

With v0.16.0 there is also a small helper [cron-like package](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.16.3/tools/cron) that could be used for scheduling jobs programmatically:

```go
c := cron.New()
c.MustAdd("dailyReport", "0 0 * * *", func() { ... })
c.Start()
```

_I'll move the issue to discussion in case someone have a better idea how to handle it._

**Reply by andreas-bergstrom** [reply]
Ah! That cron-package will probably do for now, thanks.

A "workflow builder" like AWS Step Functions/Application Composer would probably be what to aim for instead, in the long run.
